library/alloc/src/ffi/c_str.rs

```diff
@@ -815,6 +815,30 @@ impl From<Vec<NonZero<u8>>> for CString {
     }
 }
 
+impl FromStr for CString {
+    type Err = NulError;
+
+    /// Converts a string `s` into a [`CString`].
+    ///
+    /// This method is equivalent to [`CString::new`].
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
+}
+
+impl TryInto<String> for CString {
```

Comment:
This should be `impl TryFrom<CString> for String`; we don't implement `Into` and `TryInto` directly.